File: Hospital.c/src/pacientes.c
#include <stdio.h>
#include <stdlib.h>
#include "pacientes.h"

#define MaxNome 100

union document
{
    char cpf[15];
    char rg[12];
};


struct pacientes
{
    char nome[MaxNome];
    Document documento;
    char enfermidade [50];
    int internado; //sim == 1, nao == 0
    char receita [50];
};

struct hospital
{
    int codigo;
    int leitos;
    char localizacao[20];
    char nome[15];
    Pacientes paciente;
};

File: Hospital.py/src/pacientes.py
class Document:
    def __init__(self, cpf="", rg=""):
        self.cpf = cpf
        self.rg = rg

class Paciente:
    def __init__(self, nome="", documento=None, enfermidade="", internado=0, receita=""):
        self.nome = nome
        self.documento = documento if documento is not None else Document()
        self.enfermidade = enfermidade
        self.internado = internado
        self.receita = receita

class ListaPacientes:
    def __init__(self, paciente=Paciente()):
        self.paciente = paciente
        self.next = None
        self.prev = None       